biosimulators_utils/report/io.py

```diff
@@ -267,12 +267,21 @@ def run(self, report, base_path, rel_path, format=ReportFormat.h5):
                         data_set_shape = data_set_shapes[i_data_set]
                         data_set_slice = [slice(0, dim_len) for dim_len in data_set_shape] + \
                             [slice(0, 1)] * (data_set_ndim - len(data_set_shape))
-                        results[data_set.id] = (
-                            data_set_results[i_data_set][data_set_slice]
-                            .reshape(data_set_shape)
-                            .astype(data_set_data_type)
-                        )
-
+                        if data_set_ndim == 2:
+                            results[data_set.id] = (
+                                data_set_results[i_data_set]
+                                [data_set_slice[0].start:data_set_slice[0].stop,
+                                    data_set_slice[1].start:data_set_slice[1].stop]
+                                .reshape(data_set_shape)
+                                .astype(data_set_data_type)
+                            )
+                        else:
```

Comment:
it is not clear to me (without digging) if data_set_ndim will always be either 1 or 2.  I would suggest either an d-dimensional solution, or make the else: an elif data_set_ndim == 1, followed by a final else: which raises an exception that there is an unexpected number of dimensions.

Now, it will only slice the first dimension if data_set_ndim != 2 ... could we get data_set_ndim > 2 ???

Reply:
You are right, I should make this more general. The original solution using a `slice` object was general, but I'm not yet sure how to re-create that. I'll look into it.